### PR TITLE
Rerun experiments enhancement

### LINKED ISSUE #230: Rerun experiments re-runs on all previous versions of model/protocol instead of just latest
When uploading a new version of a model/protocol the 're-run experiments using this version' is really nice, but it is re-running all the old protocols/models and not just the latest one, which is probably what we want the default behaviour to be.

### PR DESCRIPTION
**Pull request by sroderick-g5sro**
Fixes #230

### CODE REVIEW

File: weblab/entities/processing.py [hunk 1]
```diff
@@ -171,21 +170,24 @@ def record_experiments_to_run(user, entity, commit):
...
+            if other_entity.is_version_visible_to_user(other_version.sha, user):
                 # Record the new experiment to run
                 kwargs = {
                     'submitter': user,
-                    'model_id': expt.model_id,
-                    'model_version': expt.model_version.sha,
-                    'protocol_id': expt.protocol_id,
-                    'protocol_version': expt.protocol_version.sha,
+                    entity.other_type: other_entity,
+                    other_entity + '_version': other_version.sha,
```
Comment (sroderick-g5sro) [inline weblab/entities/processing.py:189]: This line errors with `TypeError: unsupported operand type(s) for +: 'ProtocolEntity' and 'str'`. I wonder if the fix is as simple as wrapping other_entity in str()?

Comment (jonc125) [reply weblab/entities/processing.py:189]: Ah, no, I think it should be `entity.other_type + '_version'` sorry!

File: weblab/entities/processing.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def process_check_protocol_callback(data):
 def record_experiments_to_run(user, entity, commit):
     """Record what experiments to run automatically on a new entity version.
 
-    Any experiments that were run with the parent version(s) (that are visible to the user)
+    The latest experiments that were run with the parent version(s) (that are visible to the user)
```
Comment (jonc125) [inline weblab/entities/processing.py:163]: I think it's worth expanding this description slightly to be more detailed, based on what's in https://github.com/ModellingWebLab/WebLab/issues/230#issuecomment-705484207. Would you agree?

File: weblab/entities/tests/test_views.py [hunk 3]
```diff
@@ -1418,22 +1419,24 @@ def _add_experiment(proto_author, proto_vis, shared=False,
             }), content_type='application/json')
             assert response.status_code == 200
             detail = json.loads(response.content.decode())
+            print(detail)
```
Comment (jonc125) [inline weblab/entities/tests/test_views.py:1422]:
```suggestion
```

File: weblab/entities/tests/test_views.py [hunk 4]
```diff
@@ -1418,22 +1419,24 @@ def _add_experiment(proto_author, proto_vis, shared=False,
...
             assert 'updateEntityFile' in detail
+
```
Comment (jonc125) [inline weblab/entities/tests/test_views.py:1424]:
```suggestion
```

**Review by jonc125 — APPROVED** [review]
Looks good! I've suggested a few minor style/documentation tweaks. No need to get a re-review once those are done though.

**Pull request merged**